### PR TITLE
Add support for French decade with century

### PR DESCRIPTION
**Pull request by MichaelMWW**
Add support for recognizing French decade with century. It now supports Arabic numbers and language specific number words, examples as below.
1. dans les années 90
2. les années quatre-vingt-dix

### CODE REVIEW

**Review by tellarin — CHANGES_REQUESTED** [review]
Don't forget to always add (or remove the "NotSupportted" attribute) samples for the cases directly affected by the PR to the corresponding DateTimeModel.json. This will ensure the final output is indeed correct and in the correct format.

File: .NET/Microsoft.Recognizers.Definitions.Common/French/DateTimeDefinitions.cs [hunk 1]
```diff
@@ -43,9 +43,9 @@ public static class DateTimeDefinitions
       public const string WrittenOneToNineRegex = @"(?:une?|deux|trois|quatre|cinq|six|sept|huit|neuf)";
       public const string WrittenElevenToNineteenRegex = @"(?:(seize|quinze|quatorze|treize|douze|onze)|dix\W(neuf|huit|sept))";
       public const string WrittenTensRegex = @"(?:quatre\Wvingt(s|\Wdix)?|soixante(\Wdix)?|dix|vingt|trente|quarante|cinquante|septante|octante|huitante|nonante)";
-      public static readonly string WrittenCenturyFullYearRegex = $@"(?:(deux\s+)?mille((\s+{WrittenOneToNineRegex})?\s+cents?)?)";
+      public static readonly string WrittenCenturyFullYearRegex = $@"(?:(deux\s+)?mille((\s+{WrittenOneToNineRegex})?\s+cent?)?)";
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Definitions.Common/French/DateTimeDefinitions.cs:46]: Why this change? The s? is already saying the s is optional, useful for cases where you could have a duration of multiple centuries.

File: .NET/Microsoft.Recognizers.Definitions.Common/French/DateTimeDefinitions.cs [hunk 2]
```diff
@@ -720,11 +721,20 @@ public static class DateTimeDefinitions
       public const string NightRegex = @"\b(minuit|nuit)\b";
       public static readonly Dictionary<string, int> WrittenDecades = new Dictionary<string, int>
         {
-            { @"", 0 }
+            { @"cent", 0 },
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Definitions.Common/French/DateTimeDefinitions.cs:724]: Why is this falling in the decades config?

File: .NET/Microsoft.Recognizers.Definitions.Common/French/DateTimeDefinitions.cs (hunk 1 above)
Comment (MichaelMWW) [reply .NET/Microsoft.Recognizers.Definitions.Common/French/DateTimeDefinitions.cs:46]: > Why this change? The s? is already saying the s is optional, useful for cases where you could have a duration of multiple centuries.

In French, when express century and decade, cents is incorrect, it should be cent.

File: .NET/Microsoft.Recognizers.Definitions.Common/French/DateTimeDefinitions.cs (hunk 2 above)
Comment (MichaelMWW) [reply .NET/Microsoft.Recognizers.Definitions.Common/French/DateTimeDefinitions.cs:724]: In other language like English, the hundreds is also in the WrittenDecades.

**Comment by MichaelMWW** [comment]
> Don't forget to always add (or remove the "NotSupportted" attribute) samples for the cases directly affected by the PR to the corresponding DateTimeModel.json. This will ensure the final output is indeed correct and in the correct format.

OK.

File: .NET/Microsoft.Recognizers.Definitions.Common/French/DateTimeDefinitions.cs (hunk 1 above)
Comment (MichaelMWW) [reply .NET/Microsoft.Recognizers.Definitions.Common/French/DateTimeDefinitions.cs:46]: > > Why this change? The s? is already saying the s is optional, useful for cases where you could have a duration of multiple centuries.
> 
> In French, when express century and decade, cents is incorrect, it should be cent.

Since s is optional, it should be fine to keep it, I will revert the change.

File: .NET/Microsoft.Recognizers.Definitions.Common/French/DateTimeDefinitions.cs (hunk 2 above)
Comment (MichaelMWW) [reply .NET/Microsoft.Recognizers.Definitions.Common/French/DateTimeDefinitions.cs:724]: The "cent" entry is unnecessary, I will remove it.

Comment (MichaelMWW) [reply .NET/Microsoft.Recognizers.Definitions.Common/French/DateTimeDefinitions.cs:724]: > In other language like English, the hundreds is also in the WrittenDecades.

The "cent" entry is unnecessary, I will remove it.

**Review by aurghob — APPROVED** [review]
LGTM

**Review by tellarin — APPROVED** [review]
